prefect_github/credentials.py

```diff
@@ -27,6 +27,33 @@ class GitHubCredentials(Block):
         default=None, description="A GitHub personal access token (PAT)."
     )
 
+    def get_client(self) -> HTTPEndpoint:
+        """
+        Gets an authenticated GitHub GraphQL HTTPEndpoint client.
+
+        Returns:
+            An authenticated GitHub GraphQL HTTPEndpoint client.
+
+        Example:
+            Gets an authenticated GitHub GraphQL HTTPEndpoint client.
+            ```python
+            from prefect_github import GitHubCredentials
+
+            github_credentials = GitHubCredentials(token=token)
+            client = github_credentials.get_client()
+            ```
+        """
+
+        if self.token is not None:
+            base_headers = {"Authorization": f"Bearer {self.token.get_secret_value()}"}
+        else:
+            base_headers = None
+
+        endpoint = HTTPEndpoint(
+            "https://api.github.com/graphql", base_headers=base_headers
+        )
+        return endpoint
+
     def get_endpoint(self) -> HTTPEndpoint:
```

Comment:
Since `get_endpoint` is the same as `get_client`, do you think that we should deprecate `get_endpoint`?